app/models/exercise.rb

```diff
@@ -1,5 +1,6 @@
 class Exercise < ApplicationRecord
-  has_and_belongs_to_many :routines
+  has_many :exercise_routines
```

Comment:
Se você tentar excluir uma rotina que possui um registro na tabela `exercise_routines` tu vai encontrar um problema comum nesse tipo de relação. Dê uma olhada nessa documentação para solucionar o mesmo.

https://guides.rubyonrails.org/association_basics.html#options-for-belongs-to-dependent